Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs

```diff
@@ -31,11 +76,22 @@ void OnSelect()
                 if (placing)
                 {
                     SpatialMappingManager.Instance.DrawVisualMeshes = true;
+
+                    Debug.Log(gameObject.name + " : " + "Removing existing world anchor if any.");
```

Comment:
Someone may change this code and not call CreateAnchor() in the AnchorStoreReady function. Which means you are starting your components without adding anchors and only adding them when placement is done. So "if any" just means that a developer may not have an anchor attached the object previously but we will attempt to destroy it anyway.

DestroyImmediate(null) works fine. Internal code specifics are hard to say from their docs but I would imagine it's a safe assumption to make that they do. At least I don't see any null reference exceptions :)